src/test/java/io/r2dbc/pool/ConnectionPoolUnitTests.java

```diff
@@ -152,4 +154,83 @@ void shouldCreateMultipleConnections() {
         verify(connectionFactoryMock).create();
         assertThat(createCounter).hasValue(2);
     }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    void shouldTimeoutCreateConnection() {
+
+        ConnectionFactory connectionFactoryMock = mock(ConnectionFactory.class);
+        Connection connectionMock = mock(Connection.class);
+        when(connectionFactoryMock.create()).thenReturn((Publisher) Mono.defer(() ->
+                Mono.delay(Duration.ofMillis(100)).map(ignore -> connectionMock))
+        );
+
+        ConnectionPoolConfiguration configuration = ConnectionPoolConfiguration.builder(connectionFactoryMock)
+                .initialSize(0)
+                .maxCreateConnectionTime(Duration.ofMillis(10))
+                .build();
+        ConnectionPool pool = new ConnectionPool(configuration);
+
+
+        pool.create().as(StepVerifier::create)
+                .expectSubscription()
+                .expectError(TimeoutException.class)
+                .verify();
+
+        verify(connectionFactoryMock).create();
```

Comment:
We should verify in some sense that the connection remains in the pool after hitting a timeout. Probably reactor-pool does the right thing, but these features are likely to produce leaks.

Reply:
Added a new test method that checks connection after timeout comes from pooled one.